library/modules/DFSDL.cpp

```diff
@@ -124,3 +135,33 @@ void DFSDL::DFSDL_FreeSurface(SDL_Surface *surface) {
 int DFSDL::DFSDL_PushEvent(SDL_Event *event) {
     return g_SDL_PushEvent(event);
 }
+
+std::string DFSDL::DFSDL_GetClipboardTextUtf8() {
+    if (g_SDL_HasClipboardText() != SDL_TRUE)
+        return "";
+    char *text = g_SDL_GetClipboardText();
+    std::string ret = text;
+    g_SDL_free(text);
+    return ret;
+}
+
+std::string DFSDL::DFSDL_GetClipboardTextCp437() {
+    std::string utf8text = DFSDL_GetClipboardTextUtf8();
+    return UTF2DF(utf8text);
+}
```

Comment:
I think we should keep this module to stubs around the SDL functions. Any additional logic can go in a separate module (MiscUtils, maybe?)

Reply:
that sounds fair. will do.

Reply:
though I think keeping the encapsulation of `g_SDL_free` would probably still be a good idea

Reply:
Oh, because you're changing the return type to `std::string`. I think even that conversion could go into another module. Ideally we could just generate the stubs for the SDL functions we need, and any logic changes (including type conversions) would make that hard.

Reply:
My concern is that we expose ourselves to memory leaks by leaving the call to `SDL_free` up to the client.

Reply:
We can leak memory by calling all sorts of low-level C functions. I would argue that the client should be well aware of the need to free memory. For convenience, _most_ clients should use a higher-level wrapper that handles memory for them. Same reason most client code should use std::string's `operator+` instead of `strcat()`.